WalletWasabi/WabiSabi/Models/DecompositionAlgs/RestrictedSetDecomposer.cs

```diff
@@ -0,0 +1,56 @@
+using NBitcoin;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Models.DecompositionAlgs
+{
+	public class RestrictedSetDecomposer
+	{
+		private Decomposition Decomposition { get; } = new();
+
+		public RestrictedSetDecomposer()
+		{
+			Decomposition.Extend(new Coin() { Amount = Money.Zero });
+		}
+
+		public virtual Money? PickNextAmount(Money remaining, Money dustThreshold)
+		{
+			return LargestDenomBelowIncl(remaining)?.Amount;
+		}
+
+		public void Decompose(Money amount, Money dustThreshold, FeeRate feeRate)
+		{
+			Money remaining = amount - feeRate.GetFee(Coin.OutputVbytes);
+
+			while (remaining > dustThreshold)
+			{
+				Money? denom = PickNextAmount(remaining, dustThreshold);
+				if (denom is null || denom == Money.Zero)
+				{
+					break;
+				}
+				Coin coin = new() { Amount = denom };
+				Decomposition.Extend(coin);
+				remaining -= coin.EffectiveCost(feeRate);
+			}
+
+			if (remaining > dustThreshold)
```

Comment:
It cannot enter here because otherwise it would have not exited from the while.

Reply:
`PickNextAmount` can return null so it can happen:

https://github.com/zkSNACKs/WalletWasabi/pull/5822/files/6b49bee49a9aa6f73813c21a47aac88359c5ba7f#diff-eb41b36d007a7ee1ec5924c68a2a775c378fdf24b545a8fa63c480cf7582995aR27-R29

Reply:
Fixed that.